test/replication-luatest/gh_8121_memtx_mvcc_replication_stream_test.lua

```diff
@@ -0,0 +1,68 @@
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group(nil, t.helpers.matrix({engine = {'memtx', 'vinyl'}}))
+
+g.before_all(function(cg)
+    cg.replica_set = replica_set:new()
+    cg.master = cg.replica_set:build_and_add_server(
+            {
+                alias = 'master',
+                box_cfg = {
+                    replication_timeout = 0.1,
+                },
+            })
+    cg.replica = cg.replica_set:build_and_add_server(
+            {
+                alias = 'replica',
+                box_cfg = {
+                    replication = server.build_listen_uri('master'),
+                    replication_timeout = 0.1,
+                    memtx_use_mvcc_engine = true,
+                    txn_isolation = 'read-confirmed',
+                }
+            })
+    cg.replica_set:start()
+    cg.master:exec(function(engine)
+        local s = box.schema.space.create('s', {engine = engine})
+        s:create_index('pk')
+        s:insert{0}
+    end, {engine = cg.params.engine})
+    t.helpers.retrying({}, function()
+        cg.replica:assert_follows_upstream(cg.master:get_instance_id())
+    end)
+    cg.master:wait_for_downstream_to(cg.replica)
+    cg.replica:exec(function()
+        box.cfg{replication = ""}
+    end)
+end)
+
+g.after_all(function(cg)
+    cg.replica_set:drop()
+end)
+
+-- Checks that replication stream does not get transaction conflict errors.
+g.test_replication_stream_transaction_conflict_errors = function(cg)
+    cg.replica:exec(function()
+        local t = require('luatest')
+
+        t.assert_equals(box.space.s:select{}, {{0}})
+    end)
+    cg.master:exec(function()
+        box.space.s:delete{0}
```

Comment:
Could you please add a comment describing why there're two deletes?

Reply:
Actually, this is just a reproducer from the ticket, but I also added a comment as you asked for.